pkg/guacrest/server/retrieveDependencies.go

```diff
@@ -320,75 +335,80 @@ func mapPkgNodesToPurls(ctx context.Context, gqlClient graphql.Client,
 		logger.Warnf("GQL query \"nodes\" did not return the expected number of results")
 	}
 
-	// map the package tries to purls
-	purls := make([]string, 0, len(gqlNodes.GetNodes()))
+	// map the package tries to a map of "pkg id : purl"
+	purlMap := make(map[string]string)
 	for _, gqlNode := range gqlNodes.GetNodes() {
 		if v, ok := gqlNode.(*gql.NodesNodesPackage); ok {
 			purl := assembler_helpers.AllPkgTreeToPurl(&v.AllPkgTree)
-			purls = append(purls, purl)
+			purlMap[v.AllPkgTree.Namespaces[0].Names[0].Versions[0].Id] = purl
 		} else {
 			logger.Warnf("Nodes query returned an unexpected type: %T", *gqlNode.GetTypename())
 		}
 	}
-	return purls, nil
+	return purlMap, nil
 }
 
-/********* The endpoint handler *********/
-func (s *DefaultServer) RetrieveDependencies(
+// GetDepsForPackage gets all direct and transitive dependencies for a given purl
+func GetDepsForPackage(
 	ctx context.Context,
-	request gen.RetrieveDependenciesRequestObject,
-) (gen.RetrieveDependenciesResponseObject, error) {
+	gqlClient graphql.Client,
+	purl string,
+) (map[string]string, error) {
 	// Find the start node
-	var start node
-	if request.Params.Purl != nil {
-		pkg, err := helpers.FindPackageWithPurl(ctx, s.gqlClient, *request.Params.Purl)
-		if err != nil {
-			return handleErr(ctx, err), nil
-		}
-		start = &pkg
-	} else if request.Params.Digest != nil {
-		artifact, err := helpers.FindArtifactWithDigest(ctx, s.gqlClient, *request.Params.Digest)
-		if err != nil {
-			return handleErr(ctx, err), nil
-		}
-		start = &artifact
-	} else {
-		return gen.RetrieveDependencies400JSONResponse{
-			BadRequestJSONResponse: gen.BadRequestJSONResponse{
-				Message: "Neither a purl or a digest argument was provided",
-			}}, nil
+	unescapedPurl, err := url.QueryUnescape(purl)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unescape package url: %w", err)
+	}
+	pkg, err := helpers.FindPackageWithPurl(ctx, gqlClient, unescapedPurl)
```

Comment:
This doesn't implement the "partial purl" functionality documented in the API spec. See the other comments on this, but at least document in the spec that it isn't yet supported.